com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -22,151 +24,105 @@ public SwizzleNode()
         const string kInputSlotName = "In";
         const string kOutputSlotName = "Out";
 
-        public override bool hasPreview
-        {
-            get { return true; }
-        }
-
-        public sealed override void UpdateNodeAfterDeserialization()
-        {
-            AddSlot(new DynamicVectorMaterialSlot(InputSlotId, kInputSlotName, kInputSlotName, SlotType.Input, Vector4.zero));
-            AddSlot(new Vector4MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector4.zero));
-            RemoveSlotsNameNotMatching(new[] { InputSlotId, OutputSlotId });
-        }
-
-        static Dictionary<TextureChannel, string> s_ComponentList = new Dictionary<TextureChannel, string>
-        {
-            {TextureChannel.Red, "r" },
-            {TextureChannel.Green, "g" },
-            {TextureChannel.Blue, "b" },
-            {TextureChannel.Alpha, "a" },
-        };
-
         [SerializeField]
-        TextureChannel m_RedChannel;
+        string _maskInput = "xyzw";
 
-        [ChannelEnumControl("Red Out")]
-        public TextureChannel redChannel
+        [TextControl("Mask:")]
+        public string maskInput
         {
-            get { return m_RedChannel; }
+            get { return _maskInput; }
             set
             {
-                if (m_RedChannel == value)
+                if (_maskInput.Equals(value))
                     return;
-
-                m_RedChannel = value;
-                Dirty(ModificationScope.Node);
+                _maskInput = value;
+                UpdateNodeAfterDeserialization();
+                owner.ValidateGraph();
+                Dirty(ModificationScope.Topological);
             }
         }
 
-        [SerializeField]
-        TextureChannel m_GreenChannel;
-
-        [ChannelEnumControl("Green Out")]
-        public TextureChannel greenChannel
+        //1.mask(xyzw) 0< length <=4
+        //2.mask cant be character other than "xyzw"
+        //3.If the input is not valid, wont genetate shader code and give errors
+        public bool ValidateMaskInput(int InputValueSize)
         {
-            get { return m_GreenChannel; }
-            set
+            bool MaskInputIsValid = true;
+            char[] MaskChars = _maskInput.ToCharArray();
+            char[] AllChars  = {'x', 'y' , 'z', 'w'};
+            List<char> CurrentChars = new List<char>();
```

Comment:
Instead of copying from the AllChars arrayr into the CurrentChars list, you can simply initialize the CurrentChars list to be what you want:
List<char> CurrentChars = new List<char>() { 'x', 'y', 'z', 'w' };